### ISSUE TITLE
Throw error when `--lxc-conf` is used with native driver

### ISSUE THREAD

**Issue by candlerb** [docker/docker#11385]
To replicate: run the docker daemon with `-b none` but *not* with any `-e` flag.

~~~
root@kit1:~# docker run -d --name pctest --lxc-conf="lxc.network.type=veth" \
  --lxc-conf="lxc.network.link=docker0" --lxc-conf="lxc.network.flags=up" phusion/baseimage:latest
a7ab640b333d2f7a3e5859dad272fb43e32fd98cf72d7e60abb396ba7ae0deb2
root@kit1:~# docker exec pctest ifconfig eth0
eth0: error fetching interface information: Device not found
~~~

Expected behaviour: I think that docker run should reject any `--lxc-conf` option if the daemon is not using the lxc driver.

Test based on [this posting](https://goldmann.pl/blog/2014/01/30/assigning-ip-addresses-to-docker-containers-via-dhcp/). Further discussion [here](https://groups.google.com/forum/#!topic/docker-dev/hvs58JQ1vDs)

(Aside: the lxc configuration is accepted when running docker with `-e lxc`, but then `docker exec` does not work with that driver!)

**Comment by LK4D4** [comment]
Sorry, but `--lxc-conf` works only with `lxc` execdriver.

**Comment by candlerb** [comment]
I think you missed the point.

I do not expect the option to work.

What I ecpect is an error to be shown if I try to use the lxc-conf option with the native driver. At the option is accepted.

**Comment by jfrazelle** [comment]
I changed the name and added white label as this would be an easy fix.

**Comment by Mashimiao** [comment]
Hello,
I sent a pull #11401, trying to fix this issue

**Comment by jfrazelle** [comment]
Closed by #11401
